src/actions/retriveAlreadyBookedDaysActions.js

```diff
@@ -1,77 +1,89 @@
-
-const parseJSONFromFile = path => {
-    return readFile(path)
-        .then(parseJSON)
-        .catch(err => {
-            throw {
-                status: 500,
-                message: "the error was logged and we’ll be checking it shortly"
-            };
-        });
-};
-
-const findBookedDays = (userId, daysOff) => {
-    let daysBooked = [];
-    daysOff.forEach(element => {
-        if (element["userId"] === userId) {
-            daysBooked.push(...element["daysOff"]);
-        }
-    });
-    return daysBooked;
-};
+const DaysOff = require("../models/daysOffModel")
+const User = require("../models/userModel")
+const findBookedDays = (userId) => {
+    return new Promise((resolve, reject) => {
+        findDaysOff = DaysOff.find({ userId: userId }, "daysOff", (err, daysOff) => {
+            if (err) reject(err)
+            resolve(JSON.parse(JSON.stringify(daysOff)))
+        })
+    })
+}
 
-const getAlreadyBookedDays = (userId, daysOff) => {
-    return new Promise((response, reject) => {
-        const bookedDays = findBookedDays(userId, daysOff);
-        if (bookedDays.length === 0)
+const checkUserExistsInDB = (userId) => {
+    return new Promise((resolve, reject) => {
+        let findUser = User.findById(userId, (err, user) => {
+            if (err)
+                reject({
+                    status: 400,
+                    message: "Bad user id"
+                });
+            if (user) resolve();
             reject({
                 status: 404,
-                message: "User not found/no days booked yet"
+                message: "User not found"
             });
-        else {
-            response(bookedDays);
-        }
+        });
     });
 };
 
+const createArrayOffBookedDaysOff = (allBookedDays) => {
+    if (allBookedDays.length === 0) throw {
+        status: 404,
+        message: "No days booked yet"
+    }
+    let alreadyBookedDays = []
+    for (i of allBookedDays) {
+        alreadyBookedDays.push(...i.daysOff)
+    }
+    return alreadyBookedDays
+}
+
+const getAlreadyBookedDays = (userId) => {
+    return findBookedDays(userId)
+        .then(createArrayOffBookedDaysOff)
+        .then(data => { return data })
+        .catch(err => { throw err })
+}
+
 const sendResponse = (request, response, bookedDays) => {
     return new Promise((resolve, reject) => {
         try {
             response.status(200).send({
                 bookedDays: bookedDays,
                 links: {
                     POST: `http://localhost:3000/days`,
-                    DELETE: `http://localhost:3000/days/${Number(
+                    DELETE: `http://localhost:3000/days/${
                         request.params.id
-                    )}`
-                }
-            });
-            resolve();
+                        }`,
+                },
+            })
+            resolve()
         } catch (err) {
-            reject(err);
+            reject(err)
```

Comment:
[experiment] What would happen if the code in send response throws an error? Who catches it? What error would that be?